app/controllers/appointments_controller.rb

```diff
@@ -5,22 +5,27 @@ def index
 	end
 
 	def new
-		@patient = Patient.all
-		@doctor = Doctor.all
+		@patients = Patient.all
+		@doctors = Doctor.all
 		@appointment = Appointment.new
 	end
 
 	def create
 		modified_params= appointment_params.merge(doctor_id: params[:doctor_id], patient_id: params[:patient_id])
 		@appointment=Appointment.new(modified_params)
-		respond_to do |format|
-	      if @appointment.save
-	        format.html { redirect_to admin_index_path, notice: 'Appointment was successfully created.' }
-	        format.json { render :show, status: :created, location: @appointment }
-	      else
-	        format.html { render :new }
-	        format.json { render json: @appointment.errors, status: :unprocessable_entity }
-	      end
+		if Appointment.overload(@appointment.doctor_id,@appointment.schedule)
```

Comment:
we could inject the dependency simply by passing the appointment object, like `Appointment.overload(@appointment)`. However, this should have been an instance method as `if @appointment.overloaded?`.  In any case, I don't think either of those is an acceptable solution. It is the doctor who is overloaded for today, not the appointment. So, it could either be `Doctor.new(@appointment.doctor_id).full?` or `@appointment.bookable?`. I'd go with the latter.